src/rpc.rs

```diff
@@ -49,13 +49,14 @@ pub async fn handle_rpc_call(rpc_call: RpcCall) -> Result<Response, Rejection> {
     // Get the auth token if possible
     let auth_token = get_auth_token(&rpc_call);
     // Get the room ID
-    let room_id = get_room_id(&rpc_call);
+    let room_id_str =
+        Some(get_room_id(&rpc_call).ok_or(Error::ValidationFailed)?.get_id().to_string());
```

Comment:
This changes the semantics of the original code actually. It was possible (and some requests probably rely on this) to not include the room_id at all if it wasn't required for some particular call (otherwise `handle_get_request` wouldn't need to accept `Option` in the first place). By doing the above, you are actually combining "no room id" with "invalid room id", whereas they should probably be handled differently.